tensorflow_lite_support/python/test/task/audio/audio_embedder_test.py

```diff
@@ -0,0 +1,128 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for audio_embedder."""
+
+import enum
+
+from absl.testing import parameterized
+# TODO(b/220067158): Change to import tensorflow and leverage tf.test once
+# fixed the dependency issue.
+import unittest
+
+from tensorflow_lite_support.python.task.core.proto import base_options_pb2
+from tensorflow_lite_support.python.task.processor.proto import embedding_options_pb2
+from tensorflow_lite_support.python.task.audio import audio_embedder
+from tensorflow_lite_support.python.task.audio.core import tensor_audio
+from tensorflow_lite_support.python.test import base_test
+from tensorflow_lite_support.python.test import test_util
+
+
+_BaseOptions = base_options_pb2.BaseOptions
+_AudioEmbedder = audio_embedder.AudioEmbedder
+_AudioEmbedderOptions = audio_embedder.AudioEmbedderOptions
+
+_YAMNET_EMBEDDING_MODEL_FILE = 'yamnet_embedding_metadata.tflite'
+
+
+class ModelFileType(enum.Enum):
+  FILE_CONTENT = 1
+  FILE_NAME = 2
+
+
+class AudioEmbedderTest(parameterized.TestCase, base_test.BaseTestCase):
+
+  def setUp(self):
+    super().setUp()
+    self.model_path = test_util.get_test_data_path(_YAMNET_EMBEDDING_MODEL_FILE)
+
+  @parameterized.parameters(
+    (_YAMNET_EMBEDDING_MODEL_FILE, False, False, ModelFileType.FILE_NAME, 1024,
+     1),
+    (_YAMNET_EMBEDDING_MODEL_FILE, True, True, ModelFileType.FILE_CONTENT, 1024,
+     1)
+  )
+  def test_embed(self, model_name, l2_normalize, quantize, model_file_type,
+                 embedding_length, expected_similarity):
+    # Create embedder.
+    model_path = test_util.get_test_data_path(model_name)
+    if model_file_type is ModelFileType.FILE_NAME:
+      base_options = _BaseOptions(file_name=model_path)
+    elif model_file_type is ModelFileType.FILE_CONTENT:
+      with open(model_path, "rb") as f:
+        model_content = f.read()
+      base_options = _BaseOptions(file_content=model_content)
+    else:
+      # Should never happen
+      raise ValueError("model_file_type is invalid.")
+
+    options = _AudioEmbedderOptions(
+      base_options,
+      embedding_options_pb2.EmbeddingOptions(
+        l2_normalize=l2_normalize, quantize=quantize))
+    embedder = _AudioEmbedder.create_from_options(options)
+
+    # Load the input audio files.
+    tensor0 = tensor_audio.TensorAudio.create_from_wav_file(
+      test_util.get_test_data_path("speech.wav"),
+      embedder.required_input_buffer_size)
+
+    tensor1 = tensor_audio.TensorAudio.create_from_wav_file(
```

Comment:
Using the same file to compute similarity for now. Let me know if you'd like to change this to something else @ziyeqinghan

Reply:
Updated tests to use two different audio clips and assert cosine similarity.

Reply:
@kinaryml I saw that you're still using "speech.wav" for both tensor0 and tensor1. Did you forget to push the latest commit?